homeassistant/util/color.py

```diff
@@ -417,14 +417,59 @@ def color_rgb_to_rgbw(r: int, g: int, b: int) -> tuple[int, int, int, int]:
 
 def color_rgbw_to_rgb(r: int, g: int, b: int, w: int) -> tuple[int, int, int]:
     """Convert an rgbw color to an rgb representation."""
-    # Add the white channel back into the rgb channels.
+    # Add the white channel to the rgb channels.
     rgb = (r + w, g + w, b + w)
 
     # Match the output maximum value to the input. This ensures the
     # output doesn't overflow.
     return _match_max_scale((r, g, b, w), rgb)  # type: ignore
 
 
+def color_rgb_to_rgbww(
+    r: int, g: int, b: int, min_mireds: int, max_mireds: int
+) -> tuple[int, int, int, int, int]:
+    """Convert an rgb color to an rgbww representation."""
+    # Find the color temperature when both white channels have equal brightness
+    mired_range = max_mireds - min_mireds
+    mired_midpoint = min_mireds + mired_range / 2
+    color_temp_kelvin = color_temperature_mired_to_kelvin(mired_midpoint)
+    w_r, w_g, w_b = color_temperature_to_rgb(color_temp_kelvin)
```

Comment:
Possible alternatives would be to do either of:
 - Use the color temperature closest to RGB whitepoint at ~6500K
 - Find the color temperature best matching the given RGB color supported by the light